src/lib/encoder.ts

```diff
@@ -60,30 +57,33 @@ const bytesNRegex = /Bytes(\d+)/;
 const ALLOWED_BYTES_SIZES = [2, 4, 8, 16, 32, 64, 128, 256];
 
 const encodeDataSourceWithHash = (
-  hashType: SUPPORTED_VERIFICATION_FUNCTIONS,
-  dataHash: string,
+  verification: undefined | Verification,
   dataSource: string,
 ): string => {
-  const verificationFunction = getVerificationFunction(hashType);
-
+  if (!verification || !verification.method || !verification.data) {
+    throw new TypeError('No verification function or data provided.');
+  }
+  const verificationMethod = getVerificationMethod(verification.method);
   return (
-    keccak256(verificationFunction.name).slice(0, 10) +
-    dataHash.slice(2) +
+    keccak256(verificationMethod.name).slice(0, 10) +
+    verification.data.slice(2) +
```

Comment:
Suggest using method from web3.js for readability

```suggestion
    stripHexPrefix(verification.data) +
```

Reply:
ok